pkg/cloud/aws/services/ec2/instances.go

```diff
@@ -228,6 +228,10 @@ func (s *Service) createInstance(machine *actuators.MachineScope, bootstrapToken
 			}
 
 			userData, err = userdata.NewJoinControlPlane(&userdata.ControlPlaneJoinInput{
+				AdditionalFiles: append(
+					s.scope.ClusterConfig.AdditionalUserDataFiles,
```

Comment:
Do we write the cluster back when we close the scope? I'm wondering if this will add the machine's config to the cluster's.

Reply:
It won't, `append` creates a new slice from the one passed in

Reply:
It makes a new slice, but the backing array may be the same if it's large enough: https://play.golang.org/p/VWESAzrxGmB

Reply:
I guess that would go the other way, though – unless we change the len of the `scope.ClusterConfig.AdditionalUserDataFiles` we won't write back the changes, but future appends to that slice could clobber what we've handed off to the `ControlPlaneJoinInput`.

I guess in this case it's ok, since it immediately consumes the data and puts it into a string. Still might be safer to make a new slice from both :man_shrugging:

Reply:
I think it's safe to use with append, I don't think there is any case where this slice has cap != len